src/core/Akka/Actor/Futures.cs

```diff
@@ -148,10 +148,8 @@ public static Task<T> Ask<T>(this ICanTell self, Func<IActorRef, object> message
                 ctr2 = cancellationToken.Register(() => result.TrySetCanceled());
             }
 
-            //create a new tempcontainer path
-            var path = provider.TempPath();
-
-            var future = new FutureActorRef<T>(result, path, provider);
```

Comment:
Moved these two lines to the factory method - all of the TCS setup is left as-is.